src/ansible_navigator/configuration_subsystem/utils.py

```diff
@@ -0,0 +1,174 @@
+"""Utilities related to the configuration subsystem."""
+
+import sys
+
+from types import SimpleNamespace
+from typing import Dict
+from typing import List
+from typing import NamedTuple
+from typing import Union
+
+from .definitions import ApplicationConfiguration
+from .definitions import CliParameters
+from .definitions import Constants as C
+from .definitions import SettingsEntry
+
+
+class HumanReadableEntry(NamedTuple):
+    """Data structure for a setting entry."""
+
+    #: The possible values
+    choices: List
+    #: The CLI parameters, short and long
+    cli_parameters: Dict[str, str]
+    #: The path to the current settings file
+    current_settings_file: str
+    #: The current, effective value
+    current_value: Union[bool, Dict, str, List]
+    #: The default value
+    default: Union[bool, str]
+    #: A short description
+    description: str
+    #: The environment variable
+    env_var: str
+    #: Indicate if the current == the default
+    is_default: bool
+    #: The name
+    name: str
+    #: A sample settings file snippet
+    settings_file_sample: Union[str, Dict]
+    #: The source of the current value
+    source: str
+
+
+def transform_settings(
+    settings: ApplicationConfiguration,
+) -> List[Dict[str, Union[bool, Dict, str, List]]]:
+    """Transform the current settings into a list of dictionaries.
+
+    :param settings: The current settings
+    :returns: The settings represented as a list of dictionaries
+    """
+    settings_list = []
+
+    entry = _settings_file_entry(settings.internals)
+    settings_list.append(entry._asdict())
+
+    for current in settings.entries:
+        application_name = settings.application_name
+        settings_file_path = settings.internals.settings_file_path
+        entry = _standard_entry(
+            current=current,
+            application_name=application_name,
+            settings_file_path=settings_file_path,
+        )
+        # py36, py37 dataclass._asdict returns OrderedDict
+        if sys.version_info >= (3, 8):
+            entry_as_dict = entry._asdict()
+        else:
+            entry_as_dict = dict(entry._asdict())
+        settings_list.append(entry_as_dict)
+
+    sorted_settings = sorted(settings_list, key=lambda d: d["name"])
+    return sorted_settings
+
+
+def _sample_generator(settings_path: str) -> Dict[str, Union[Dict, str]]:
```

Comment:
Could you use a verb to describe what action this performs? Does it really return a generator as the function name suggests?

Reply:
sample creator would be a viable alternative.  I don't believe generator here was intended to refer to a python generator but instead the act of generating something, in this case, a sample settings file entry.

I also think it is clear by the return type, a generator is not returned.

Reply:
Also, "sample" could be perceived as a verb too. So there's that...

Reply:
settings_file_sample_creator

Reply:
But it's not a creator, it's not a static object/class, it's a function so start its name with what it does.
```suggestion
def _create_sample_settings_file(settings_path: str) -> Dict[str, Union[Dict, str]]:
```

Reply:
This suggests it's creating a sample setting file when it is really creating a settings file sample.

Reply:
So is this better?
```suggestion
def _create_settings_file_sample(settings_path: str) -> Dict[str, Union[Dict, str]]:
```

Reply:
Awesome.

Reply:
done in next commit